Products/CMFPlone/browser/syndication/views.py

```diff
@@ -50,6 +55,62 @@ def __call__(self):
             return self.index()
 
 
+class NewsMLFeedView(FeedView):
+
+    def context_enabled(self):
+        settings = IFeedSettings(self.context, None)
+        if settings and not settings.enabled:
+            raise NotFound
+        else:
+            return True
+
+    @lazy_property
+    def current_date(self):
+        return DateTime()
+
+    def duid(self, item, value):
+        uid = uuid3(NAMESPACE_OID, item.uid + str(value))
+        return uid.hex
+
+    def get_image(self, item):
+        scales = item.context.restrictedTraverse('@@images')
+        if scales:
+            try:
+                return scales.scale('image')
+            except AttributeError:
+                pass
+        return None
+
+    def newsml_allowed(self):
+        util = getMultiAdapter((self.context, self.request),
+                               name='syndication-util')
+        if not util.site_enabled():
+            return False
+        elif ISyndicatable.providedBy(self.context):
+            settings = IFeedSettings(self.context, None)
+            if settings.enabled:
+                return True
+        return False
+
+    def newsml_enabled(self, raise404=False):
+        if not self.newsml_allowed():
+            if raise404:
+                raise NotFound
+            else:
+                return False
+        else:
+            return True
+
+    def __call__(self):
```

Comment:
Why is the __call__ method overridden here? I don't think newsml_allowed and newsml_enabled are then necessary.